torchbiggraph/train_cpu.py

```diff
@@ -770,11 +801,35 @@ def _load_embeddings(
             optimizer.load_state_dict(optim_state)
         return embs, optimizer
 
+    def _write_single_embedding(
+        self,
+        holder: EmbeddingHolder,
+        entity: EntityName,
+        part: Partition):
+        embs = holder.partitioned_embeddings[(entity, part)]
+        optimizer = self.trainer.partitioned_optimizers[(entity, part)]
+        self.checkpoint_manager.write(
+            entity, part, embs.detach(), optimizer.state_dict()
+        )
+
+    def _nondestructive_write_embedding(self, bucket: Bucket):
+        parts: Set[Tuple[EntityName, Partition]] = set()
+        parts.update((e, bucket.lhs) for e in self.holder.lhs_partitioned_types)
+        parts.update((e, bucket.rhs) for e in self.holder.rhs_partitioned_types)
+        for entity, part in parts:
+            self._write_single_embedding(self.holder, entity, part)
+    
+    def _write_stats(self, bucket: Optional[Bucket], stats: Optional[BucketStats]):
```

Comment:
This naming is misleading... I think it does more than write stats in the distributed scheduler.